x/pairing/keeper/staking.go

```diff
@@ -56,6 +56,29 @@ func (k Keeper) StakeNewEntry(ctx sdk.Context, validator, creator, chainID strin
 			utils.Attribute{Key: "geolocation", Value: geolocation},
 		)
 	}
+
+	// validate there are no more than 5 endpoints per geolocation
+	endp_geo_hist := map[int32]int{}
```

Comment:
do it much more simple:
if len(enpoints) > len(geolocations)*5
     error

Reply:
this is a different restriction.
a user can have 6 endpoints in US and one in EU - thus bypassing the 5 endpoints per geolocation restriction but passing this if

Reply:
done. see https://github.com/lavanet/lava/commit/0f853db7a5ebf23bf8455e136b67aa18ff476d8e